hack/collapse/test_auto_collapse.sh

```diff
@@ -0,0 +1,26 @@
+#!/bin/bash
+
+SCRIPT_DIR=$( cd -- "$( dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )
+
+# Paths to files
+export INPUT_FILE="$SCRIPT_DIR/test_example.md"
+export OUTPUT_FILE="$SCRIPT_DIR/test_example_collapsible.md"
+export EXPECTED_FILE="$SCRIPT_DIR/test_example_expected_collapsible.md"
+export SCRIPT="$SCRIPT_DIR/auto_collapse.sh"
+
+# Run the script
+bash "$SCRIPT" "$INPUT_FILE" "$OUTPUT_FILE" 5
+
+if [ ! -f "$OUTPUT_FILE" ]; then
+    echo "Test failed: Output file not found."
```

Comment:
nitpicking but interesting: When a string is passed in `"`, bash will look into the string if there is anything to expand. Using `'` will tell bash to not look into the string because we do not want anything to expand. Thus, using `'` for plain strings is a bit more performant (not relevant for this script)